src/core/lbgpu_cuda.cu

```diff
@@ -284,6 +284,15 @@ __device__ T xyz_to_index(T* xyz){
   return x + para.dim_x*(y + para.dim_y*z);
 }
 
+template < typename T >
+__device__ void index_to_xyz_LE(T index, T* xyz){
```

Comment:
if possible dont use ouput variables

Reply:
I don't really get what you mean. Are you referring to the xyz?

Reply:
yes